src/main/python/hydra/lib/runtestbase.py

```diff
@@ -267,6 +336,23 @@ def reset_all_app_stats(self, name):
             ha_sub.reset_stats()
             ha_sub.stop()  # closes the ANalyser socket, can not be used anymore
 
+    def reset_app_group_stats(self, name, g_index):
```

Comment:
enhance the existing tasks
"reset_all_app"
"ping_all_app"
"refresh_app_info"
etc to accept a list of application ID's such that you can re-use that code.